crates/matrix-sdk/src/media.rs

```diff
@@ -542,45 +542,41 @@ impl Media {
                     thumbnail_source,
                     thumbnail_info,
                 });
-                let mut image_message_event_content =
-                    ImageMessageEventContent::plain(body, url).info(Box::new(info));
-                image_message_event_content.filename = filename;
-                image_message_event_content.formatted = config.formatted_caption;
-                MessageType::Image(image_message_event_content)
+                let mut content = ImageMessageEventContent::plain(body, url).info(Box::new(info));
+                content.filename = filename;
+                content.formatted = config.formatted_caption;
```

Comment:
Could also use the `assign!` macro here (I think I saw it still being used in a bunch of other places). Same for the other changes in this commit.

Reply:
Yep, `make_attachment` in the latest version does maximally use `assign!`, IMO :+1: